pkg/signatures/engine/engine.go

```diff
@@ -344,3 +367,36 @@ func (engine *Engine) GetSelectedEvents() []detect.SignatureEventSelector {
 	}
 	return res
 }
+
+func (engine *Engine) RegisterDataSource(dataSource detect.DataSource) error {
+	engine.dataSourcesMutex.Lock()
+	defer engine.dataSourcesMutex.Unlock()
+
+	namespace := dataSource.Namespace()
+	id := dataSource.ID()
+
+	if _, ok := engine.dataSources[namespace]; !ok {
+		engine.dataSources[namespace] = map[string]detect.DataSource{}
+	}
```

Comment:
So what happens if we register same data source twice ? Or a diff data source with the same namespace ? Could you document the namespace directive, what it means, obligations of callers, etc ? Thanks!

Reply:
diff data source in the same namespace gets registered to that namespace. ID collision in a namespace returns an error.
I'll document it in the godoc.